vernac/declare.mli

```diff
@@ -437,6 +437,16 @@ val declare_constant
   -> constant_entry
   -> Constant.t
 
+(** Like [declare_definition] but also returns the final UState.t *)
+val declare_definition_full
+  :  info:Info.t
+  -> cinfo:EConstr.t option CInfo.t
+  -> opaque:bool
+  -> body:EConstr.t
+  -> ?using:Vernacexpr.section_subset_expr
+  -> Evd.evar_map
+  -> GlobRef.t * UState.t
```

Comment:
Not sure the ustate is what we really want to return here, maybe the universes_entry (1st component of `univs` in prepare_definition)  is better

Reply:
(also maybe the hooks want that too instead of the minimized-and-restricted ustate, but that's probably not for this PR)